packages/main/src/utils/Strings.js

```diff
@@ -0,0 +1,65 @@
+/**
+ * Escape a regular expression text so that it can be used in a regular expression.
+ * @param {string} text The raw text
+ * @returns The regular expression text, with special characters having been escaped.
+ */
+export function escapeRegex(text) {
+  return text.replace(/[.*+?^${}()|[\]\\]/g, "\\$&");
+}
+
+/**
+ * Replace all instances of a particular find text.
+ * @param {string} text The text to find/replace on
+ * @param {string} find The string to find
+ * @param {string|(s:string)=>string} replace The replace string or function
+ * @param {boolean=} caseInsensitive true if the replace is case insensitive, false by default
+ * @return {string} The string with all matches replaced.
+ */
+export function replaceAll(text, find, replace, caseInsensitive) {
+  return text.replace(new RegExp(escapeRegex(find), `${caseInsensitive ? 'i' : ''}g`), replace);
+}
+
+// Cache the regular expressions for repeated use
+const markupReplace = `&=&amp,>=&gt;,<=&lt;,"=&quot;,'=&#39;`.split(",")
+  .map((p) => p.split('='))
+  .map(([f, r]) => [new RegExp(escapeRegex(f), 'g'), r]);
+
+/**
+ * Escape all html special characters in a string to their html entity equivalent.
+ * @param {string} text The raw text, which may contain markup
+ * @return {string} The escaped text
+ */
+export function escapeHTML(text) {
+  markupReplace.forEach(([regex, replacement]) => {
```

Comment:
the new `replaceAll` function could be used here as well, but I don't mind explicitly passing the `'g'` flag as is currently the case

Reply:
The RegExp objects are "cached" in the `markupReplace` array whereas the `replaceAll` function would be re-creating the same RegExp object over and over. So I think keeping it this way will be a little bit more efficient.